### PR TITLE
Disable SIGPIPE signals when the client disconnects

### PR DESCRIPTION
**Pull request by nil-ableton**
When running Remotery on MacOS I see SIGPIPE signals
emitted when playing around with the browser based client
(refreshing it for instance)

When this signal is not handled by an application this leads
to a crash.

Solution:

- BSD, we set the `SO_NOSIGPIPE` flag using `setsockopt`
- Linux, we pass the `MSG_NOSIGNAL` flag to `send`

Tested on MacOSX 10.11

### CODE REVIEW

**Comment by nil-ableton** [comment]
References:
For linux, http://linux.die.net/man/2/send
For BSD, https://www.freebsd.org/cgi/man.cgi?query=setsockopt&sektion=2

File: lib/Remotery.c [hunk 1]
```diff
@@ -1885,7 +1897,13 @@ static rmtError TCPSocket_Send(TCPSocket* tcp_socket, const void* data, rmtU32 l
...
     {
         // Attempt to send the remaining chunk of data
-        int bytes_sent = (int)send(tcp_socket->socket, cur_data, (int)(end_data - cur_data), 0);
+        int send_flags = 0;
+#ifdef MSG_NOSIGNAL
+        // On Linux this prevents send from emitting a SIGPIPE signal
+        // Equivalent on BSD to the SO_NOSIGPIPE option.
+        send_flags = MSG_NOSIGNAL;
+#endif
+        int bytes_sent = (int)send(tcp_socket->socket, cur_data, (int)(end_data - cur_data), send_flags);
```
Comment (dwilliamson) [inline lib/Remotery.c]: This won't compile in C mode.

File: lib/Remotery.c [hunk 2]
```diff
@@ -1885,7 +1897,13 @@ static rmtError TCPSocket_Send(TCPSocket* tcp_socket, const void* data, rmtU32 l
     while (cur_data < end_data)
     {
         // Attempt to send the remaining chunk of data
-        int bytes_sent = (int)send(tcp_socket->socket, cur_data, (int)(end_data - cur_data), 0);
+        int send_flags = 0;
+#ifdef MSG_NOSIGNAL
+        // On Linux this prevents send from emitting a SIGPIPE signal
+        // Equivalent on BSD to the SO_NOSIGPIPE option.
+        send_flags = MSG_NOSIGNAL;
```
Comment (dwilliamson) [inline lib/Remotery.c]: This won't compile in C mode.

**Comment by dwilliamson** [comment]
Thanks, again. Can you adjust your tabs to 4xSpaces?

**Comment by nil-ableton** [comment]
I've corrected the spacing and moved the variable declarations at the top of the scope. Good catch :)

**Comment by dwilliamson** [comment]
Good job. Thanks for the thorough explanation.

**Pull request merged**